### PR TITLE
Add new file menu entries for different file types.

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
App can register new file handlers

## Related Issue
- #274 

## Screenshots (if appropriate):
Screenshot from 2019-10-01 12-42-53
Screenshot from 2019-10-01 12-43-12

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5364/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5385/

<p>






</p></details>

**Comment by PVince81** [comment]
there are conflicts to fix

File: apps/files/src/mixins.js [hunk 1]
```diff
@@ -457,6 +457,25 @@ export default {
...
+        const url = window.location.origin + '/' + path
+        const target = `${action.routeName}-${filePath}`
+        const win = window.open(url, target)
+        // in case popup is blocked win will be null
+        if (win) {
+          win.focus()
+        }
+        return
+      }
+      // TODO: rewire code below ....
```
Comment (PVince81) [inline apps/files/src/mixins.js]: raise tickets for these ?

File: apps/files/src/components/FilesAppBar.vue [hunk 2]
```diff
@@ -349,17 +353,25 @@ export default {
         p.then(() => {
           this.createFile = false
           this.$_ocFilesFolder_getFolder()
+          this.fileFolderCreationLoading = false
+          if (this.newFileAction) {
+            // not cool - needs refactoring
+            this.$nextTick(() => {
+              this.openFile({
```
Comment (PVince81) [inline apps/files/src/components/FilesAppBar.vue]: I'm confused about what the difference is between `openFile` and `openFileAction` and also why we call both after each other in a synchronous manner.

Please rename the functions to more explicit names or add comments here.

Comment (DeepDiver1975) [reply apps/files/src/components/FilesAppBar.vue]: Part of the tech dept - refs TODO above ... will open ticket

Comment (PVince81) [reply apps/files/src/components/FilesAppBar.vue]: ref https://github.com/owncloud/phoenix/issues/2300

Comment (DeepDiver1975) [reply apps/files/src/components/FilesAppBar.vue]: https://github.com/owncloud/phoenix/issues/2300

**Pull request merged**